pkg/drivers/vfkit/driver_darwin.go

```diff
@@ -97,20 +98,25 @@ func (d *Driver) getDiskPath() string {
 	return d.ResolveStorePath(fmt.Sprintf("%s.img", d.MachineName))
 }
 
-func (d *Driver) resize(newSize int64) error {
+func (d *Driver) resize(newSize uint64) error {
+	if newSize > math.MaxInt64 {
+		return fmt.Errorf("integer overflow detected for resize: %v", newSize)
+	}
+	//#nosec G115 -- os.stat provide the size as int64 and os.Truncate use int64 as arg. Also range is checked (above)
```

Comment:
`#nosec G115` is not needed since you added the range check. You can keep the rest of the comment if you think it's useful.